test/classes.coffee

```diff
@@ -471,13 +471,6 @@ test "`new` shouldn't add extra parens", ->
   ok new Date().constructor is Date
 
 
-test "`new` works against bare function", ->
-
-  eq Date, new ->
-    eq this, new => this
```

Comment:
So the 1.x output for this is:
```js
eq(Date, new function() {
  eq(this, new ((function(_this) {
    return function() {
      return _this;
    };
  })(this)));
  return Date;
});
```
The output per this PR is:
```js
eq(Date, new function() {
  eq(this, new (() => {
    return this;
  }));
  return Date;
});
```
The latter block doesn’t throw any compilation errors per the Babel REPL, but throws `TypeError: (intermediate value) is not a constructor` in the Node runtime. See [MDN: TypeError: "x" is not a constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Not_a_constructor). Per [MDN: Arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), arrow functions “can not be used as constructors.” (Not to educate you @connec, since you obviously know this, but just for others who might find this thread.)

So if by definition you can’t use `new` with an arrow function, I don’t think we need to write a test that proves that our runtime enforces JavaScript rules. One could argue that we should throw an error in the compiler if someone tries `new =>`, but we don’t currently check or throw errors for all the prohibited constructors, like `new Math`.

There’s nothing wrong with keeping just the first line though, and the last `Date` line. That at least preserves the test that `new` works against regular functions. I’ll restore that.